src/reporter.ts

```diff
@@ -89,23 +95,33 @@ const scrubSpan = (span) => {
         span.info.httpInfo.response.body = scrub(
           decodeHttpBody(response.body, isError),
           response.headers,
+          ScrubContext.HTTP_RESPONSE_BODY,
           sizeLimit,
           span.info.httpInfo.response.truncated
         );
       }
       if (span.info.httpInfo.request?.body) {
         span.info.httpInfo.request.body = scrub(
           decodeHttpBody(request.body, isError),
-          request.headers,
+          response.headers,
```

Comment:
Is this intended to scrub the response headers and not the request as it did previously?

Reply:
That's a big, fat bug you just found

Reply:
The impact actually was that, if the response did not have a JSON-like MIME type, but the request was, we might fail to scrub the latter. Thanks for catching it.